src/oauth2.rs

```diff
@@ -263,23 +270,62 @@ pub async fn dc_get_oauth2_addr(
 }
 
 impl Oauth2 {
-    fn from_address(addr: impl AsRef<str>) -> Option<Self> {
+    async fn from_address(addr: impl AsRef<str>) -> Option<Self> {
         let addr_normalized = normalize_addr(addr.as_ref());
         if let Some(domain) = addr_normalized
             .find('@')
             .map(|index| addr_normalized.split_at(index + 1).1)
         {
-            match domain {
-                "gmail.com" | "googlemail.com" => Some(OAUTH2_GMAIL),
-                "yandex.com" | "yandex.by" | "yandex.kz" | "yandex.ru" | "yandex.ua" | "ya.ru"
-                | "narod.ru" => Some(OAUTH2_YANDEX),
-                _ => None,
+            if let Some(provider) = Oauth2::lookup_whitelist(domain) {
+                Some(provider)
```

Comment:
Can probably use [`Option.or_else()`](https://doc.rust-lang.org/std/option/enum.Option.html#method.or_else) to avoid this `if`.

Reply:
async closures are unstable, ref rust-lang/rust#62290